plugins/uefi-dbx/fu-dbxtool.c

```diff
@@ -59,6 +60,31 @@ fu_dbxtool_get_siglist_local (const gchar *filename, GError **error)
 					     error);
 }
 
+static gboolean
+fu_dbxtool_is_live_media (void)
+{
+	gsize bufsz = 0;
+	g_autofree gchar *buf = NULL;
+	g_auto(GStrv) tokens = NULL;
+	const gchar *args[] = {
+		"rd.live.image",
+		"boot=live",
+		NULL, /* last entry */
+	};
+
+	/* get a PII-safe kernel command line */
+	if (!g_file_get_contents ("/proc/cmdline", &buf, &bufsz, NULL))
+		return FALSE;
+	if (bufsz == 0)
+		return FALSE;
+	tokens = fu_common_strnsplit (buf, bufsz - 1, " ", -1);
+	for (guint i = 0; args[i] != NULL; i++) {
+		if (g_strv_contains ((const gchar * const *) tokens, args[i]))
+			return TRUE;
+	}
+	return FALSE;
+}
+
```

Comment:
I think this function is actually useful to all of fwupd and should go into the daemon.  For example we can use it as a hint that we shouldn't run UEFI capsule updates.

Reply:
Something like `fu_common_is_live_media()` in `libfwupdplugin/fu-common.c`?

Reply:
Yeah that's what I was thinking.  Then plugins can mark devices not updatable based upon that.
I'm also not sure if that argument list is sufficient for Ubuntu 20.04+ with their casper stuff.  I know they used to use `boot=casper` in 19.10 and earlier, but I don't know for sure what happens in 20.04 and htat will need double checking.
@xnox can you comment?

Reply:
It's best to check for presence of `/cdrom/.disk/info` to detect "installer media"

Reply:
Useful, thanks. Will update the patch.

Reply:
Fixed.